src/lib/services/CommandLineSupport.ts

```diff
@@ -32,37 +43,19 @@ export abstract class CommandLineSupport extends AsyncCreatable {
 	}
 
 	/**
-	 * Accepts a child process created by child_process.spawn(), and a Promise's resolve and reject functions.
-	 * Resolves/rejects the Promise once the child process finishes.
-	 * @param cp
-	 * @param res
-	 * @param rej
+	 * Perform any job-specific processing on the results of a child process execution, and either resolve or reject as
+	 * needed.
+	 * @param args
+	 * @protected
 	 */
-	protected monitorChildProcess(cp: ChildProcessWithoutNullStreams, res: (string) => void, rej: (string) => void): void {
-		let stdout = '';
-		let stderr = '';
-		this.getSpinnerManager().startSpinner();
-
-		// When data is passed back up to us, pop it onto the appropriate string.
-		cp.stdout.on('data', data => {
-			stdout += data;
-		});
-		cp.stderr.on('data', data => {
-			stderr += data;
-		});
+	protected abstract handleResults(args: ResultHandlerArgs): void;
 
-		cp.on('exit', code => {
-			this.parentLogger.trace(`monitorChildProcess has received exit code ${code}`);
-			if (this.isSuccessfulExitCode(code)) {
-				this.getSpinnerManager().stopSpinner(true);
-				res(stdout);
-			} else {
-				// If we got any other error, it means something actually went wrong. We'll just reject with stderr for
-				// the ease of upstream error handling.
-				this.getSpinnerManager().stopSpinner(false);
-				rej(stderr);
-			}
-		});
+	protected defaultResultHandler(args: ResultHandlerArgs): void {
```

Comment:
Default implementation of abstract method. Most of the implementers will just call `super.defaultResultHandler()`.

Reply:
This feels a little awkward to me since this abstract method's implementations are invoking another protected non-final method, which is at the same level as the abstract method's definition. Could possibly lead to cyclic invocations if `defaultResultHandler()` is overridden.

A few common patterns we could consider in this case:
1. (Composition over inheritance)[https://en.wikipedia.org/wiki/Composition_over_inheritance] - this default implementation could be in a class outside of this hierarchy
2. [Creating hooks](https://www.patterns.dev/posts/hooks-pattern/) to call additional steps before/after the default steps are done